### PR TITLE
Fix via waypoints APIs naming

### PR DESCRIPTION
**Pull request by Guardiola31337**
- Follow up from https://github.com/mapbox/mapbox-java/pull/959

We found in https://github.com/mapbox/mapbox-navigation-android/pull/1733#discussion_r255541722 that naming could be confusing and should be more explicit in order to make clients' life easier.

### CODE REVIEW

**Comment by osana** [comment]
@Guardiola31337 
If we are doing this name change, we should do a similar name change in MapboxMapMatching
I guess we can deprecate the use of `waypoints` and add `viawaypoints`  (for now)

Also please `viawaypoints` to RouteOptions

File: services-directions/src/main/java/com/mapbox/api/directions/v5/MapboxDirections.java [hunk 1]
```diff
@@ -424,7 +424,7 @@ public static Builder builder() {
     private Point destination;
     private Point origin;
     private String[] approaches;
-    private Integer[] waypoints;
```
Comment (akitchen) [inline services-directions/src/main/java/com/mapbox/api/directions/v5/MapboxDirections.java]: This was correct as-was. Should be `waypoints` as one word, no capital P

Comment (Guardiola31337) [reply services-directions/src/main/java/com/mapbox/api/directions/v5/MapboxDirections.java]: Yeah, you're correct. Will update in a follow-up PR. Thanks for flagging 🙇

**Comment by Guardiola31337** [comment]
@osana 
> If we are doing this name change, we should do a similar name change in MapboxMapMatching
I guess we can deprecate the use of `waypoints` and add `viawaypoints`  (for now)

Yeah, I agree. We should be consistent across the different services. Will implement ☝️ if we're not cool breaking SemVer.

> Also please add `viawaypoints` to RouteOptions

You mean, adding "waypoints" option to Map Matching route options, right? I've already added it to Directions 👀 https://github.com/mapbox/mapbox-java/blob/57e8710c6f1628df04277ac1978000cfad459141/services-directions/src/main/java/com/mapbox/api/directions/v5/models/RouteOptions.java#L284 https://github.com/mapbox/mapbox-java/blob/57e8710c6f1628df04277ac1978000cfad459141/services-directions/src/main/java/com/mapbox/api/directions/v5/models/RouteOptions.java#L569

**Comment by osana** [comment]
@Guardiola31337 Please add a call to viaWaypoints()  when RouteOptions are being saved/generated: https://github.com/mapbox/mapbox-java/blob/6e6110a303f7704524734022e066ef5e8d90982a/services-directions/src/main/java/com/mapbox/api/directions/v5/MapboxDirections.java#L235

I believe tests might be updated as well.

**Pull request merged**